File: src/Domain/Entities/Accountant/Transaction.cs
using System;

namespace Domain.Entities.Accountant;

public class Transaction : Entity
{
    public int Id { get; set; }
    public int? FromAccountId { get; set; }
    public int? ToAccountId { get; set; }
    public int? CategoryId { get; set; }
    public decimal Amount { get; set; }
    public decimal? FromStocks { get; set; }
    public decimal? ToStocks { get; set; }
    public string Currency { get; set; }
    public string Description { get; set; }
    public DateTime Date { get; set; }
    public bool IsEncrypted { get; set; }
    public byte[] EncryptedDescription { get; set; }
    public byte[] Salt { get; set; }
    public byte[] Nonce { get; set; }

    public Account FromAccount { get; set; }
    public Account ToAccount { get; set; }
    public Category Category { get; set; }
}

File: src/Accountant/accountant2/src/lib/models/server/requests/transaction.ts
export class CreateTransaction {
	constructor(
		public fromAccountId: number | null,
		public toAccountId: number | null,
		public categoryId: number | null,
		public amount: number,
		public fromStocks: number | null,
		public toStocks: number | null,
		public currency: string,
		public description: string | null,
		public date: string,
		public isEncrypted: boolean,
		public encryptedDescription: string | null,
		public salt: string | null,
		public nonce: string | null,
		public createdDate: Date | null,
		public modifiedDate: Date | null
	) {}
}

export class UpdateTransaction {
	constructor(
		public id: number,
		public fromAccountId: number | null,
		public toAccountId: number | null,
		public categoryId: number | null,
		public amount: number,
		public fromStocks: number | null,
		public toStocks: number | null,
		public currency: string,
		public description: string | null,
		public date: string,
		public isEncrypted: boolean,
		public encryptedDescription: string | null,
		public salt: string | null,
		public nonce: string | null,
		public createdDate: Date | null,
		public modifiedDate: Date | null
	) {}
}
